core/iwasm/common/wasm_runtime_common.c

```diff
@@ -694,7 +698,9 @@ wasm_runtime_destroy_internal(void)
     thread_manager_destroy();
 #endif
 
+#if WASM_ENABLE_INVOKE_NATIVE != 0
```

Comment:
IIUC `WASM_ENABLE_INVOKE_NATIVE` is designed to prevent the involvement of runtime built-in functions. In that case, `-DWAMR_BUILD_LIBC_BUILTIN=0 -DWAMR_BUILD_LIBC_WASI=0` can achieve the same effect, along with `-DWAMR_BUILD_QUICK_AOT_ENTRY=0`.

Reply:
i guess it's a good opportunity to replace the [heavy ifdef conditions](https://github.com/bytecodealliance/wasm-micro-runtime/blob/a6a9f1f45d9f7ebf044ceac71bcf7a9ea2f90f23/core/iwasm/common/wasm_native.c#L484-L490) with something easier to maintain.